src/main/java/com/blamejared/controlling/client/gui/GuiNewKeyBindingList.java

```diff
@@ -87,31 +83,70 @@ public int getListWidth() {
         return super.getListWidth() + 32;
     }
 
+    public List<IGuiListEntry> getAllEntries() {
+        return allEntries;
+    }
+
+    public void setDisplayedEntries(List<IGuiListEntry> displayedEntries) {
+        this.displayedEntries.clear();
+        this.displayedEntries.addAll(displayedEntries);
+    }
+
+    private void drawHighlightedString(String text, int x, int y, boolean highlight) {
+        final int index = text.toLowerCase().indexOf(controlsScreen.getSearchString().toLowerCase());
+        if (highlight && index > -1) {
+            final int indexEndHighlight = index + controlsScreen.getSearchString().length();
+            final String textStart = text.substring(0, index);
+            final String textMiddle = text.substring(index, indexEndHighlight);
+            final String textEnd = text.substring(indexEndHighlight);
+            final int rectLeft = x + mc.fontRenderer.getStringWidth(textStart);
+            final int rectRight = rectLeft + mc.fontRenderer.getStringWidth(textMiddle);
+            final int rectBottom = y + mc.fontRenderer.FONT_HEIGHT;
+            Gui.drawRect(rectLeft, y, rectRight, rectBottom, YELLOW_HIGHLIGHT_COLOR);
+            mc.fontRenderer.drawStringWithShadow(textStart, x, y, 0xFFFFFF);
+            x += mc.fontRenderer.getStringWidth(textStart);
+            mc.fontRenderer.drawString(textMiddle, x, y, DARK_TEXT_HIGHLIGHT_COLOR);
+            x += mc.fontRenderer.getStringWidth(textMiddle);
+            mc.fontRenderer.drawStringWithShadow(textEnd, x, y, 0xFFFFFF);
```

Comment:
You don't need to recalculate these coordinates:
```suggestion
            mc.fontRenderer.drawString(textMiddle, rectLeft, y, DARK_TEXT_HIGHLIGHT_COLOR);
            mc.fontRenderer.drawStringWithShadow(textEnd, rectRight, y, 0xFFFFFF);
```